supportedBackends/sfs/src/main/scala/cromwell/backend/sfs/SharedFileSystem.scala

```diff
@@ -29,51 +29,44 @@ object SharedFileSystem {
 
-  type PathsPair = (Path, Path)
-  type DuplicationStrategy = (Path, Path) => Try[Unit]
+  type FilesPair = (File, File)
+  type DuplicationStrategy = (File, File) => Try[Unit]
 
-  /**
+    /**
     * Return a `Success` result if the file has already been localized, otherwise `Failure`.
     */
-  private def localizePathAlreadyLocalized(originalPath: Path, executionPath: Path): Try[Unit] = {
-    if (File(executionPath).exists) Success(Unit) else Failure(new RuntimeException(s"$originalPath doesn't exists"))
+  private def localizePathAlreadyLocalized(originalPath: File, executionPath: File): Try[Unit] = {
+    if (executionPath.exists) Success(()) else Failure(new RuntimeException(s"$originalPath doesn't exists"))
   }
 
-  private def localizePathViaCopy(originalPath: Path, executionPath: Path): Try[Unit] = {
-    File(executionPath).parent.createDirectories()
+  private def localizePathViaCopy(originalPath: File, executionPath: File): Try[Unit] = {
+    executionPath.parent.createDirectories()
     val executionTmpPath = pathPlusSuffix(executionPath, ".tmp")
-    Try(File(originalPath).copyTo(executionTmpPath, overwrite = true).moveTo(executionPath, overwrite = true))
+    Try(originalPath.copyTo(executionTmpPath, overwrite = true).moveTo(executionPath, overwrite = true))
   }
 
-  private def localizePathViaHardLink(originalPath: Path, executionPath: Path): Try[Unit] = {
-    File(executionPath).parent.createDirectories()
-    Try(Files.createLink(executionPath, originalPath))
+  private def localizePathViaHardLink(originalPath: File, executionPath: File): Try[Unit] = {
+    executionPath.parent.createDirectories()
+    Try(executionPath.linkTo(originalPath, symbolic = false)) map { _ => executionPath }
   }
 
-  /**
-    * TODO: The 'call' parameter here represents the call statement in WDL that references this path.
-    * We're supposed to not use symbolic links if the call uses Docker.  However, this is currently a
-    * bit incorrect because multiple calls can reference the same path if that path is in a declaration.
-    *
-    * The symbolic link will only fail in the Docker case if a Call uses the file directly and not
-    * indirectly through one of its input expressions
-    */
-
-  private def localizePathViaSymbolicLink(originalPath: Path, executionPath: Path): Try[Unit] = {
-      if (File(originalPath).isDirectory) Failure(new UnsupportedOperationException("Cannot localize directory with symbolic links"))
+  private def localizePathViaSymbolicLink(originalPath: File, executionPath: File): Try[Unit] = {
+      if (originalPath.isDirectory) Failure(new UnsupportedOperationException("Cannot localize directory with symbolic links"))
       else {
-        File(executionPath).parent.createDirectories()
-        Try(Files.createSymbolicLink(executionPath, originalPath.toAbsolutePath))
+        executionPath.parent.createDirectories()
+        Try(executionPath.linkTo(originalPath, symbolic = true)) map { _ => executionPath }
```

Comment:
Could you comment this? It's a little bit funky?

`"Try to create a symlink (which we ignore), and if that works return the original, otherwise return the linking failure"`?